node.cpp

```diff
@@ -0,0 +1,192 @@
+#include <mutex>
+
+#include "node.h"
+#include "roles.h"
+
+std::once_flag init_once, cleanup_once;
+
+void Node::initialize()
+{
+  std::call_once(init_once, []() { xmlInitParser(); });
+}
+
+void Node::cleanup()
+{
+  std::call_once(cleanup_once, []() { xmlCleanupParser(); });
```

Comment:
Wouldn't this be a problem if there are other `Node` instances surviving the first one calling `cleanup`? (untested, just asking).